repondeur/zam_repondeur/templates/lectures_add.html

```diff
@@ -43,48 +40,56 @@ <h2>Ajouter une lecture...</h2>
+<script src="{{ request.static_url('zam_repondeur:static/jquery/jquery-1.12.4.min.js') }}"></script>
+<script src="{{ request.static_url('zam_repondeur:static/selectize/js/standalone/selectize.min.js') }}"></script>
 <script type="text/javascript">
 
-    var lectures_by_dossier = {{ lectures_by_dossier | tojson | safe }};
     document.addEventListener("DOMContentLoaded", function() {
 
-        var dossierElem = document.getElementById("select-dossier");
-        var dossierChoices = new Choices(dossierElem, {
-            shouldSort: false,
-        });
-
-        var lectureElem = document.getElementById("select-lecture");
-        var lectureChoices;
-        updateFancySelectWidget();
-
-        dossierElem.addEventListener("change", function(event) {
-            resetFancySelectWidget();
-            updateBaseSelectOptions(event.detail.value);
-            updateFancySelectWidget();
-        }, false);
-
-        function resetFancySelectWidget() {
-            lectureChoices.destroy();
-        }
-
-        function updateBaseSelectOptions(value) {
-            lectureElem.innerHTML = '';
-            var choices = lectures_by_dossier[value];
-            for (var texte_uid in choices) {
-                var elem = document.createElement("option")
-                elem.value = texte_uid;
-                elem.text = choices[texte_uid];
-                lectureElem.appendChild(elem);
+        var selectLectures = $("#select-lecture").selectize({
+            valueField: "uid",
+            labelField: "label",
+        })[0];
+
+        $("#select-dossier").selectize({
+            valueField: "uid",
+            labelField: "titre",
+            searchField: ["titre"],
+            onInitialize: function() {
+                var self = this;
+                $.ajax({
+                    url: "{{ request.route_url('choices_dossiers') }}",
+                    type: "GET",
+                    error: function() {
+                        alert("Impossible de récupérer la liste des dossiers.\nMerci de réessayer plus tard !");
+                    },
+                    success: function(res) {
+                        for (let dossier of res.dossiers) {
+                            self.addOption(dossier);
+                        }
+                    }
+                });
+            },
+            onChange: function(uid) {
+                selectLectures.selectize.clearOptions();
+                if (uid === "") {
+                    return;
+                }
+                $.ajax({
+                    url: "{{ request.route_url('choices_lectures', uid='UID') }}".replace("UID", uid),
+                    type: "GET",
+                    error: function() {
+                        alert("Impossible de récupérer la liste des lectures.\nMerci de réessayer plus tard !");
+                    },
+                    success: function(res) {
+                        for (let uid in res.lectures) {
```

Comment:
I think that syntax is not consistent with the fact of using `var` elsewhere. We should discuss as a team which browsers/versions we want to support.